docs/usage/TYPE_SYNC.md

```diff
@@ -0,0 +1,120 @@
+# commercetools type sync
+
+Utility which provides API for building CTP type update actions and type synchronisation.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Usage](#usage)
+  - [Sync list of type drafts](#sync-list-of-type-drafts)
+    - [Prerequisites](#prerequisites)
+    - [Running the sync](#running-the-sync)
+  - [Build all update actions](#build-all-update-actions)
+  - [Build particular update action(s)](#build-particular-update-actions)
+- [Caveats](#caveats)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Usage
+        
+### Sync list of type drafts
+
+#### Prerequisites
+1. The sync expects a list of non-null `TypeDrafts` objects that have their `key` fields set to match the
+types from the source to the target. Also the target project is expected to have the `key` fields set, otherwise they won't be
+matched.
+2. It is an important responsibility of the user of the library to instantiate a `sphereClient` that has the following properties:
+    - Limits the amount of concurrent requests done to CTP. This can be done by decorating the `sphereClient` with
+   [QueueSphereClientDecorator](http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/client/QueueSphereClientDecorator.html)
+    - Retries on 5xx errors with a retry strategy. This can be achieved by decorating the `sphereClient` with the
+   [RetrySphereClientDecorator](http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/client/RetrySphereClientDecorator.html)
+
+   You can use the same client instantiating used in the integration tests for this library found
+   [here](/src/main/java/com/commercetools/sync/commons/utils/ClientConfigurationUtils.java#L45).
+
+4. After the `sphereClient` is setup, a `TypeSyncOptions` should be be built as follows:
+````java
+// instantiating a TypeSyncOptions
+final TypeSyncOptions typeSyncOptions = TypeSyncOptionsBuilder.of(sphereClient).build();
+````
+
+The options can be used to provide additional optional configuration for the sync as well:
+- `errorCallBack`
+a callback that is called whenever an event occurs during the sync process that represents an error. Currently, these
+events.
+
+- `warningCallBack`
+a callback that is called whenever an event occurs during the sync process that represents a warning. Currently, these
+events.
+
+- `beforeUpdateCallback`
+a filter function which can be applied on a generated list of update actions. It allows the user to intercept type
+update and modify (add/remove) update actions just before they are send to CTP API.
+
+- `beforeCreateCallback`
+a filter function which can be applied on a type draft before a request to create it on CTP is issued. It allows the
+user to intercept type create requests modify the draft before the create request is sent to CTP API.
+
+- `allowUuid`
```

Comment:
Will it work? I don't see you use it anywhere.

Reply:
removed from the documentation.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/304/commits/2ddf05b211f1b471e4b8a377cc20b73c8d53823b#diff-649a74fcd11f3ef15302f4486a9ddbd1L59